ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/CommandId.java

```diff
@@ -42,7 +42,8 @@
     CREATE,
     DROP,
     EXECUTE,
-    TERMINATE
+    TERMINATE,
+    ALTER
```

Comment:
hmm, I'm wondering what this means for compatibility - I'm not sure what adding a value to an enum will do to the JSON serde. I think it works but we should test issuing some commands on an earlier version of ksqlDB then upgrading to a version that has this patch and seeing that it works. Similarly, we should test what happens if we downgrade (submit something with `ALTER` in it, and then roll back). We should make sure we gracefully handle the failure and enter `DEGRADED` state)

FYI @stevenpyzhang

Reply:
I don't think there's any good way to gracefully handle the failure without changing the earlier versions, which kinda goes against the point. And on top of that, the ddl command subtype `alterSourceV1` also causes backwards compatibility issues. So given that this is supposed to just be an alternative to `CREATE OR REPLACE`, I'll go ahead and  reuse the commands for `createStream/createTable` for this instead.

Reply:
Sorry, ignore what I said - I tested this on a really old version that does not have the degraded state change. This actually does enter `DEGRADED` state, and switching between versions works.